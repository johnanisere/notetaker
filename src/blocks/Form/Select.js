import styled from "styled-components";

import Select from "../../elements/Select";

export default styled(Select)`
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background-color: rgba(0, 0, 0, 0.03);
  appearance: none;
  border: 1px solid #d4d4d4;
  padding: 12px;
  box-sizing: border-box;
  margin-top: 8px;
  border-radius: 4px;
`;
