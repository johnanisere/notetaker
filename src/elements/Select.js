import styled from "styled-components";

const Select = styled.select`
  width: 100%;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 #d7d7e0;
  border: 1px solid #d4d4d4;
  padding: 12px;
  background-image: url(https://buzzhostingservices.com/images/drop-down-arrow-7.png);
  background-size: auto 25px;
  background-color: white;
  background-position: calc(100% - 10px) center;
  background-repeat: no-repeat;
  outline: none;
  transition: outline 0.2s ease-in-out 0s;
`;

export default Select;
