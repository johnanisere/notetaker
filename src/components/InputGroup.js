import React from "react";
import Form from "../blocks/Form";
import Select from "./Select";

const Group = ({ title, value, name, onChange, type, list = [] }) => (
  <Form.Label>
    {title}
    <br />
    {type === "select" ? (
      <Select value={value} name={name} onChange={onChange} list={list} />
    ) : (
      <Form.Input value={value} name={name} type={type} onChange={onChange} />
    )}
  </Form.Label>
);

export default Group;
