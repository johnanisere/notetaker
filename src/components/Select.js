import React from "react";
import Form from "../blocks/Form";

const Select = ({ list = [], value, onChange, name }) => (
  <Form.Select onChange={onChange} value={value} name={name}>
    <option value="">Select</option>
    {list.map((value, key) => (
      <option value={value} key={key}>
        {value}
      </option>
    ))}
  </Form.Select>
);

export default Select;
