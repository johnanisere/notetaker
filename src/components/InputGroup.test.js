import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";

import InputGroup from "./InputGroup";
import BlockForm from "../blocks/Form";
import Select from "./Select";

Enzyme.configure({ adapter: new Adapter() });

it("InputGroup component exists", () => {
  const wrapper = shallow(<InputGroup />);
  expect(wrapper.exists()).toBe(true);
});

it("InputGroup component renders", () => {
  const wrapper = shallow(<InputGroup />);

  expect(wrapper).toMatchSnapshot();
});

it(`Should contain Input`, () => {
  const props = {
    title: "Note Text",
    value: "Hey",
    name: "note",
    type: "text",
    onChange: () => {}
  };
  expect.assertions(3);
  const wrapper = shallow(<InputGroup {...props} />);
  expect(wrapper.text()).toBe(props.title);
  expect(wrapper.find(BlockForm.Input)).toHaveLength(1);
  expect(wrapper.find(BlockForm.Input).props().value).toEqual(props.value);
});

it(`Should contain Select`, () => {
  const props = {
    title: "Note Text",
    value: "Hey",
    name: "note",
    type: "select",
    list: ["red", "blue", "white"],
    onChange: () => {}
  };
  expect.assertions(1);
  const wrapper = shallow(<InputGroup {...props} />);

  expect(wrapper.find(Select)).toHaveLength(1);
});
