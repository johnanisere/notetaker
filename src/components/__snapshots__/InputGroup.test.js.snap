// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Group />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        <br />,
        <ForwardRef />,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": undefined,
          "onChange": undefined,
          "type": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  width: 100%;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 #d7d7e0;
  border: 1px solid #d4d4d4;
  padding: 0 12px;
  box-sizing: border-box;
  margin-top: 8px;
  border-radius: 4px;
",
              "
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background: rgba(0, 0, 0, 0.03);
",
            ],
          },
          "displayName": "Styled(styled.input)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": true,
        "rules": Array [
          "
  font-size: 14px;
  color: rgba(0, 0, 0, 0.5);
  width: 100%;
  margin-bottom: 18px;
  font-weight: 500;
",
        ],
      },
      "displayName": "styled.label",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": "label",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          <br />,
          <ForwardRef />,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": undefined,
            "onChange": undefined,
            "type": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  width: 100%;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 #d7d7e0;
  border: 1px solid #d4d4d4;
  padding: 0 12px;
  box-sizing: border-box;
  margin-top: 8px;
  border-radius: 4px;
",
                "
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background: rgba(0, 0, 0, 0.03);
",
              ],
            },
            "displayName": "Styled(styled.input)",
            "foldedComponentIds": Array [
              "sc-bdVaJa",
            ],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "input",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "rules": Array [
            "
  font-size: 14px;
  color: rgba(0, 0, 0, 0.5);
  width: 100%;
  margin-bottom: 18px;
  font-weight: 500;
",
          ],
        },
        "displayName": "styled.label",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "label",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
